examples/shot_detection/shot_detect.py

```diff
@@ -77,85 +77,128 @@ def place_image(i, fr):
 
 
 def make_montage_scanner(db, table, shot_starts):
-    row_length = 8
-    rows_per_item = 1
-    target_width = 256
-
-    frame = table.as_op().gather(
-        shot_starts,
-        item_size = row_length * rows_per_item)
-
-    montage = db.ops.Montage(
-        frame = frame,
-        num_frames = row_length * rows_per_item,
-        target_width = target_width,
-        frames_per_row = row_length,
-        device = DeviceType.GPU)
-
-    job = Job(columns = [montage], name = 'montage_image')
-    montage_table = db.run(job, force=True)
-
-    montage_img = np.zeros((1, target_width * row_length, 3), dtype=np.uint8)
-    for _, img in montage_table.load(['montage']):
-        if len(img[0]) > 100:
-            img = np.flip(img, 2)
-            montage_img = np.vstack((montage_img, img))
     return montage_img
 
 
 def main():
-    movie_path = util.download_video() if len(sys.argv) <= 1 else sys.argv[1]
+    total_start = time.time()
+
+    #movie_path = util.download_video() if len(sys.argv) <= 1 else sys.argv[1]
+    movie_path = '/n/scanner/datasets/movies/private/kubo_and_the_two_strings_2016.mp4'
     print('Detecting shots in movie {}'.format(movie_path))
     movie_name = os.path.basename(movie_path)
 
     # Use GPU kernels if we have a GPU
     if have_gpu():
         device = DeviceType.GPU
-        scanner_montage = True
     else:
         device = DeviceType.CPU
-        scanner_montage = False
 
+    device = DeviceType.CPU
     with Database() as db:
         print('Loading movie into Scanner database...')
         s = time.time()
-        [movie_table], _ = db.ingest_videos([(movie_name, movie_path)], force=True)
+
+        ############ ############ ############ ############
+        # 0. Ingest the video into the database
+        ############ ############ ############ ############
+        [movie_table], _ = db.ingest_videos([(movie_name, movie_path)],
+                                            force=True)
         print('Time: {:.1f}s'.format(time.time() - s))
+        print('Number of frames in movie: {:d}'.format(movie_table.num_rows()))
 
         s = time.time()
+        ############ ############ ############ ############
+        # 1. Run Histogram over the entire video in Scanner
+        ############ ############ ############ ############
         print('Computing a color histogram for each frame...')
-        frame = movie_table.as_op().all()
+        frame = db.ops.FrameInput()
         histogram = db.ops.Histogram(
             frame = frame,
             device = device)
-        job = Job(columns = [histogram], name = movie_name + '_hist')
-        hists_table = db.run(job, force=True)
-        print('\nTime: {:.1f}s'.format(time.time() - s))
+        output = db.ops.Output(columns=[histogram])
```

Comment:
Is there a good reason to have an explicit output node vs. just having `BulkJob(columns=[histogram], ...)`?

Reply:
This leaves open the ability to add multiple output Ops (multiple output tables). `BulkJob(output=...)` would change to `BulkJob(outputs=[...])`